source/normalize-arguments.ts

```diff
@@ -40,8 +43,10 @@ export const preNormalizeArguments = (options: any, defaults?: any) => {
 
 	for (const event of knownHookEvents) {
 		if (is.nullOrUndefined(options.hooks[event])) {
-			if (defaults) {
-				options.hooks[event] = [...defaults.hooks[event]];
+			if (defaults && defaults.hooks) {
+				if (event in defaults.hooks) {
+					options.hooks[event] = defaults.hooks[event]!.slice();
```

Comment:
Which one is faster? `.slice()` or `[...array]`?

Reply:
If we wanted a deep clone of the defaults, we'd need something more than a `[...array]`. Otherwise, `.slice` is faster than `[...array]` from the looks of a quick benchmark